### PR TITLE
Allow for multiple queries in QA inference when using rest_api format

### PR DESCRIPTION
**Pull request by brandenchan**
This PR allows for QA inference to handle multiple queries for a single doc when using rest api mode. This addresses #242

### CODE REVIEW

File: farm/infer.py [hunk 1]
```diff
@@ -251,6 +251,13 @@ def inference_from_dicts(self, dicts, rest_api_schema=False, max_processes=128):
                 f"b) ... run inference on a downstream task: make sure your model path {self.name} contains a saved prediction head"
             )
 
+        # Assign a sequential doc_id to each dictionary
```
Comment (tholor) [inline farm/infer.py]: Shouldn't this only be executed for rest_api_schema=true? For SQuAD input format we want to use the passed doc id.

File: farm/infer.py [hunk 2]
```diff
@@ -251,6 +251,13 @@ def inference_from_dicts(self, dicts, rest_api_schema=False, max_processes=128):
...
+        old_dicts = dicts
```
Comment (tholor) [inline farm/infer.py]: Maybe I am missing something here, but why do you need the "old_dict" var here? Can't we modify the dicts in place?

File: farm/infer.py (hunk 1 above)
Comment (brandenchan) [reply farm/infer.py]: Actually there's a better way to do this now which is that the doc id for squad or rest api format is inferred from basket.id

File: farm/infer.py (hunk 2 above)
Comment (brandenchan) [reply farm/infer.py]: You're right. Change made.

File: farm/infer.py [hunk 3]
```diff
@@ -251,6 +251,11 @@ def inference_from_dicts(self, dicts, rest_api_schema=False, max_processes=128):
                 f"b) ... run inference on a downstream task: make sure your model path {self.name} contains a saved prediction head"
             )
 
+        # Assign a sequential doc_id to each dictionary
+        if rest_api_schema:
```
Comment (tholor) [inline farm/infer.py]: As I understood, we are now using baske_id instead. So we probably don't need this one anymore?

**Comment by brandenchan** [comment]
Code was tested and now document ids are part of the output when rest_api_schema=True

**Pull request merged**